Marlin/src/MarlinCore.cpp

```diff
@@ -478,6 +478,10 @@ void startOrResumeJob() {
     #ifdef EVENT_GCODE_SD_ABORT
       queue.inject_P(PSTR(EVENT_GCODE_SD_ABORT));
     #endif
+    
+    //Wait for EVENT_GCODE_SD_ABORT to finish, so it doesn't get interrupted by PRINTCOUNTER
+    planner.synchronize();
```

Comment:
This doesn't work because the `EVENT_GCODE_SD_ABORT` code is not executed before this line, but some time later in a call from the main loop.

Reply:
Thanks for looking in to it!
I tested the placement of that line and it did stop the sudden jerk when aborting, but maybe there's a randomness factor and I just got lucky?

Reply:
When a print is aborted it interrupts the current move in progress and this will lead to lost steps. It's a "feature" of the abort function, because otherwise you would have to wait for the planner buffer to empty, and it could be running several very slow moves…. The correct way to stop a print that you want to be able to restart later is to pause it and then power off the machine.

Reply:
Oh yes sorry that's not what I mean. The print still stops immediately and sometimes loses steps, it's just not as jarring as it was before. By moving print_job_timer.stop() to the bottom and adding synchronize right before that, the printer will stop immediately, disable the heaters, do the event_gcode_abort, wait for that to finish and only once all that is done it will save to eeprom. That makes absolutely sure that the printer is not moving anymore once eeprom is accessed.